### DISCUSSION TITLE
能不能把Direct按钮改成超链接？

### DISCUSSION THREAD

**Discussion by BrandonStudio** [spencerwooo/onedrive-vercel-index#296] [Ideas]
就像是 GitHub 的 Raw 按钮一样，使用 `<a>` 而不是 `<button>`，这样可以快速在新标签页打开，复制的话也可以右击复制到剪贴板

**Comment by spencerwooo** [comment]
具体哪个位置的 Direct 按钮？截个图吧

**Reply by BrandonStudio** [reply]
![image](https://user-images.githubusercontent.com/55647556/150630370-36ffdb87-8701-41e1-8aa7-0dc43f62db74.png)
和
![image](https://user-images.githubusercontent.com/55647556/150630374-90a0433a-e1e2-4922-8ee7-2ea7302659a6.png)

**Comment by spencerwooo** [comment]
试了一下，GitHub 的 `raw` 主要是可以在新标签页用浏览器直接看代码，所以用 `<a>` 合理。但是本项目的文件直链直接点击会进入下载，不会有新标签页打开预览之类的事情，所以这里链接的主要作用就是「复制到剪贴板」，也就是原交互逻辑。所以暂时不考虑修改了。

**Reply by BrandonStudio** [reply]
w
那还是得等什么时候有MIME了
……还会考虑MIME么？

**Reply by spencerwooo** [reply]
MIME 指什么？file/mimetype 吗？什么叫考虑 MIME？

**Reply by BrandonStudio** [reply]
就是 https://github.com/spencerwooo/onedrive-vercel-index/issues/130#issuecomment-932094534 里提到的MIME
我理解错了，我以为以后会考虑改用MIME而非扩展名判断文件类型
我的意思是您现在所说本项目直链点击会进入下载的原因应该是其类型是 application/octet-stream，以后是否会为不同类型的文件设置不同的MIME类型？

**Reply by spencerwooo** [reply]
类型 `application/octet-stream` 是 OneDrive 直链返回的啊，如果我进一步修改那就需要像 proxied download 那个功能一样修改返回的 header 了，而事实证明 Vercel 不乐意让你这样干（因为会走 Vercel 的流量）

**Reply by BrandonStudio** [reply]
我突然发现，直链的响应头里的 content-type 就是文件的类型，并不都是 application/octet-stream，浏览器理解为下载是因为还有一个 content-disposition 响应头。。。。。。
